src/main/java/frc/lib/io/motor/Motor.java

```diff
@@ -0,0 +1,137 @@
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.AngularAcceleration;
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Current;
+import edu.wpi.first.units.measure.Temperature;
+import edu.wpi.first.units.measure.Velocity;
+import edu.wpi.first.units.measure.Voltage;
+import frc.lib.util.CANDevice;
+
+/**
+ * Standardized interface for motor controllers used in FRC.
+ * Supports multiple control modes, telemetry reporting, and follower
+ * configuration.
+ */
+public interface Motor {
+
+    public enum PIDSlot {
+        SLOT_1,
+        SLOT_2,
+        SLOT_3;
+    }
+
+    @AutoLog
+    abstract class MotorInputs {
+        /** Whether the motor is connected. */
+        public boolean connected = false;
+        /** Motor position. */
+        public Angle position = Radians.of(0.0);
+        /** Motor velocity. */
+        public AngularVelocity velocity = RadiansPerSecond.of(0.0);
+        /** Voltage applied to the motor. */
+        public Voltage appliedVoltage = Volts.of(0.0);
+        /** Total supply current to the motor. */
+        public Current supplyCurrent = Amps.of(0.0);
+        /** Torque-producing current. */
+        public Current torqueCurrent = Amps.of(0.0);
+        /** Motor temperature in degrees. */
+        public Temperature temperature = Celsius.of(0.0);
+        /** Error in position */
+        public Angle positionError = null;
+        /** Error in velocity */
+        public AngularVelocity velocityError = null;
+        /** Active trajectory position in rotations */
+        public Angle activeTrajectoryPosition = null;
+        /** Active trajectory velocity in rotations per second. */
+        public AngularVelocity activeTrajectoryVelocity = null;
+    }
+
+    /**
+     * Returns the motor's CAN device ID.
+     *
+     * @return The CANDevice representing this motor's ID and bus name.
+     */
+    public CANDevice getID();
+
+    /**
+     * Updates the passed-in MotorInputs structure with the latest sensor readings.
+     *
+     * @param inputs Motor input structure to populate.
+     */
+    public void updateInputs(MotorInputs inputs);
+
+    /**
+     * Sets the motor to coast mode.
+     */
+    public void runCoast();
+
+    /**
+     * Sets the motor to brake mode.
+     */
+    public void runBrake();
+
+    /**
+     * Follows the specified motor using CAN follower mode.
+     *
+     * @param followMotor The motor to follow.
+     * @param oppose      Whether or not to oppose the main motor.
+     */
+    public void follow(Motor motor, boolean oppose);
+
+    /**
+     * Runs the motor using direct voltage control.
+     *
+     * @param voltage Desired voltage output.
+     */
+    public void runVoltage(Voltage voltage);
+
+    /**
+     * Runs the motor with a specified current output.
+     *
+     * @param current Desired torque-producing current.
+     */
+    public void runCurrent(Current current);
```

Comment:
Do we want to add the ability to overload this with the max dutycycle parameter that we used at somepoint last year?